### PR TITLE
Views

### PR DESCRIPTION
**Pull request by Hombre2014**
All the project requirements have been implemented.

### Project requirements
- Implement the design from the sneak peek wireframes, as shown [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md)
- Styling is nice to have but focus on making sure everything functions first.
- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).
- **Hint:** In order to see some posts or comments, you might need to add some data to your database as described [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md#use-models-to-insert-data).

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,5 +1,7 @@
-<h1>Welcome to the Ruby on Rails Blog</h1>
-<h2>Posts</h2>
-<h3>Post number 1</h3>
-<h3>Post number 2</h3>
-<h3>Post number 3</h3>
+<h2>Our members</h2>
+<% @users = User.all; %>
```
Comment (mikethreels) [inline app/views/users/index.html.erb]: - [x] the Users controller `#index` action should render all users to its corresponding Views index page.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,3 +1,19 @@
-<h1>Single user</h1>
-<h2>Bio</h2>
-<h3>Posts</h3>
+<% @users = User.all; %>
+<% @posts = Post.all; %>
+<% current_user = @users.find_by(id: params[:id]) %>
```
Comment (mikethreels) [inline app/views/users/show.html.erb]: - [x] the Users controller `#show` actions should return the user by their ID.

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+<% @posts_per_user = current_user.posts %>
+<h2><%= current_user.name %>'s page</h2>
+<div class="body">
+  <%= render partial: "user", locals: { current_user: current_user } %>
+  <div class="bio">
+    <h2>Bio</h2>
+    <p><%= current_user.bio %></p>
+  </div>
+  <h2>Last three posts</h2>
+  <% @posts_per_user.first(3).each do |post| %>
```
Comment (mikethreels) [inline app/views/users/show.html.erb]: - [x] please avoid using logic in the views, you previously created a method called `last_3_posts` in your user model to handle this.

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1,4 +1,22 @@
...
-<h3>Post number 3</h3>
+<% @users = User.all; %>
+<% current_user = @users.find_by(id: params[:user_id]) %>
+<div class="body">
+  <h2>All the posts from <%= current_user.name %></h2>
+  <div>
+    <%= render partial: "users/user", locals: { current_user: current_user } %>
+  </div>
+  <% @posts_per_user = @users.find_by(id: params[:user_id]).posts %>
+  <% @posts_per_user.first(3).each do |post| %>
```
Comment (mikethreels) [inline app/views/posts/index.html.erb]: - [x] please avoid using logic in the views, you previously created a method called `last_3_posts` in your user model to handle this.

File: app/views/posts/index.html.erb [hunk 5]
```diff
@@ -1,4 +1,22 @@
...
+    <% @comments_per_post = Post.find_by(id: post.id).comments %>
+    <%= render partial: "users/post", locals: { post: post, current_user: current_user } %>
+    <h3>Comments:</h3>
+    <div class="comments">
+      <% @comments_per_post.first(5).each do |comment| %>
```
Comment (mikethreels) [inline app/views/posts/index.html.erb]: - [x] please avoid using logic in the views, you previously created a method called `last_5_comments` in your Post model to handle this.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Yuriy, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Hombre2014** [comment]
Thanks for your recommendations and suggestions. All have been corrected.

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @Hombre2014, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**